cli/io.go

```diff
@@ -0,0 +1,205 @@
+package cli
+
+import (
+	"github.com/mongodb/grip"
+	"github.com/mongodb/grip/level"
+	"github.com/mongodb/jasper"
+	"github.com/pkg/errors"
+)
+
+// Validator represents an input that can be validated.
+type Validator interface {
+	Validate() error
+}
+
+// OutcomeResponse represents CLI-specific output describing if the request was
+// processed successfully and if not, the associated error message.  For other
+// responses that compose OutcomeResponse, their results are valid only if
+// Success is true.
+type OutcomeResponse struct {
```

Comment:
how viable is it to just json-seralize the rest models in the jasper package and use them rather than needing to have cli-package specific types?